### PR TITLE
[TM] Add Spec for DatePickerAndroid

### PR DESCRIPTION
**Pull request by pedpess**
## Summary

This PR solves part of this issue: https://github.com/facebook/react-native/issues/24875

## Changelog

[General] [Added] - TM Spec for DatePickerAndroid

## Test Plan

Flow passes.

### CODE REVIEW

**Comment by pedpess** [comment]
Hi @fkgozali :), is it normal that the e2e test fails?

**Comment by ericlewis** [comment]
@pedpess we aim for it not to be broken, but if it’s failing on iOS steps then that’s known.

**Comment by pedpess** [comment]
@ericlewis It's failing in `RNTester/e2e/__tests__/Touchable-test.js ` for EarlGrey, which is the e2e I assume we are using here? Then I think I just iOS related.

**Comment by ericlewis** [comment]
Yep, this is a known issue. You’re fine!

Sent via Superhuman iOS ( https://sprh.mn/?vip=ericlewis777@gmail.com )

**Comment by cpojer** [comment]
Hey @pedpess do you mind signing the FB CLA so we can merge your change? It only takes a minute:  code.facebook.com/cla

**Comment by pedpess** [comment]
Hi @cpojer , just did it :)

File: Libraries/Components/DatePickerAndroid/NativeDatePickerAndroid.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+'use strict';
+
+import type {TurboModule} from 'RCTExport';
+import * as TurboModuleRegistry from 'TurboModuleRegistry';
+import type {Options, DatePickerOpenAction} from './DatePickerAndroidTypes';
+import Platform from 'Platform';
+
+export interface Spec extends TurboModule {
+  +open: (options: ?Options) => Promise<DatePickerOpenAction>;
```
Comment (fkgozali) [inline Libraries/Components/DatePickerAndroid/NativeDatePickerAndroid.js]: Unfortunately some of the types in DatePickerAndroidTypes like `Date` isn't compatible yet with our codegen system, so for now let's modify these types to use `Object` (I'll modify it after I import to FB)

**Pull request closed without merging**